src/EnergyPlus/SolarShading.cc

```diff
@@ -797,6 +717,92 @@ void GetShadowingInput(EnergyPlusData &state)
           state.dataIPShortCut->cAlphaArgs(7));
 }
 
+void processShadowingInput(EnergyPlusData &state)
+{
+    // all shadow input processing that needed zones and surfaces to already be read into data (part of fix for Defect #10299)
+
+    if ((state.dataSysVars->shadingMethod == DataSystemVariables::ShadingMethod::PixelCounting) &&
+        state.dataSolarShading->anyScheduledShadingSurface) {
+        ShowSevereError(state, "The Shading Calculation Method of choice is \"PixelCounting\"; ");
+        ShowContinueError(state, "and there is at least one shading surface of type ");
+        ShowContinueError(state, "Shading:Site:Detailed, Shading:Building:Detailed, or Shading:Zone:Detailed, ");
+        ShowContinueError(state, "that has an active transmittance schedule value greater than zero or may vary.");
+        ShowContinueError(state, "With \"PixelCounting\" Shading Calculation Method, the shading surfaces will be treated as ");
+        ShowContinueError(state, "completely opaque (transmittance = 0) during the shading calculation, ");
+        ShowContinueError(state, "which may result in inaccurate or unexpected results.");
+        ShowContinueError(state, "It is suggested switching to another Shading Calculation Method, such as \"PolygonClipping\".");
+    }
+
+    if (state.dataSysVars->shadingMethod == DataSystemVariables::ShadingMethod::Imported) {
+        int ExtShadingSchedNum;
+        for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+            ExtShadingSchedNum = ScheduleManager::GetScheduleIndex(state, state.dataSurface->Surface(SurfNum).Name + "_shading");
+            if (ExtShadingSchedNum != 0) {
+                state.dataSurface->Surface(SurfNum).SurfSchedExternalShadingFrac = true;
+                state.dataSurface->Surface(SurfNum).SurfExternalShadingSchInd = ExtShadingSchedNum;
+            } else {
+                ShowWarningError(state,
+                                 format("processShadowingInput: sunlit fraction schedule not found for {} when using ImportedShading.",
+                                        state.dataSurface->Surface(SurfNum).Name));
+                ShowContinueError(state, "These values are set to 1.0.");
+            }
+        }
+    }
+
+    int SurfZoneGroup, CurZoneGroup;
+    int Found = 0;
+    if (state.dataSysVars->DisableGroupSelfShading) {
+        Array1D_int DisableSelfShadingGroups;
+        int NumOfShadingGroups = state.dataSysVars->shadingGroupsNum;
+        if (NumOfShadingGroups > 0) {
+            DisableSelfShadingGroups.allocate(NumOfShadingGroups);
+            for (int i = 1; i <= NumOfShadingGroups; i++) {
+                Found = Util::FindItemInList(state.dataSysVars->zoneName(i), state.dataHeatBal->ZoneList, state.dataHeatBal->NumOfZoneLists);
+                if (Found != 0) DisableSelfShadingGroups(i) = Found;
+            }
+
+            for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; SurfNum++) {
```

Comment:
I agree that this would be much more efficient.  I'd like to have us make a mental note of this and deal with it at a later date if you are okay with that.  I wonder if there are more places where we could be much more efficient and we don't even realize it.  If only we all had the time to do that sort of efficiency review in the code...